fix: `start_climate_control`

coreywillwhat opened the pull request:
**Fix**: `start_climate_control` service.
**Feature**: Adds `API Level` option to account for different data request structures.

See updated [README](https://github.com/audiconnect/audi_connect_ha/tree/master?tab=readme-ov-file#audi-connect-integration-for-home-assistant) for more details about `API Level`

### Important Notes
The following items are **not** currently working for climate control, but will be fixed in a future update:
- The `check_request_succeeded` method does not currently run for the EU region.
- Stop command for climate control is not yet fixed.

coreywillwhat commented:
@Kolbi @cdnninja @lexathon @shuntfield @RienduPre
Looking for testers for this PR if you can give it a shot and report back.

Kolbi commented:
For easier testing I created a manual prelease: https://github.com/audiconnect/audi_connect_ha/releases/tag/v1.11.3

lexathon commented:
I realise I set api 1 in configuration but maybe need to reload it. Will try again in an hour after I’ve dropped off the kids!

Kolbi commented:
@coreywillwhat I think we can close https://github.com/audiconnect/audi_connect_ha/pull/492 ?

shuntfield commented:
Will give a go today and report back 👍🏻

lexathon commented:
So two things. The API level change in configuration doesn't seem to retain its state - not 100% sure if switching to level 1 is being kept somewhere - URL seems right, so I'm assuming it is, but when you reopen configure, it's back at 0.

The URL matches, but I'm wondering if there's something else going on with the headers that's kicking back a forbidden?

shuntfield commented:
OK I have just tested with the pre-release, I might have gotten the wrong end of the stick. I am using the "Audi Connect: Start Climate Control" and setting a temperature on the car with my VIN number installed Action to call it, or should I be using Legacy?

I get in the log still:
Unable to start climate control of vehicle wauzzzfw8rXXXXXXXX. Error: 403, message='Forbidden', url='https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZFW8XXXXXXX/climatisation/start'

```
Traceback (most recent call last):
  File "/config/custom_components/audiconnect/audi_connect_account.py", line 320, in start_climate_control
    await self._audi_service.start_climate_control(
    ...<8 lines>...
    )
  File "/config/custom_components/audiconnect/audi_services.py", line 676, in start_climate_control
    res = await self._api.request(
          ^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
    )
    ^
  File "/config/custom_components/audiconnect/audi_api.py", line 79, in request
    raise ClientResponseError(
    ...<4 lines>...
    )
aiohttp.client_exceptions.ClientResponseError: 403, message='Forbidden', url='https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZFW8XXXXXXXX/climatisation/start'
```


On the legacy Action I get this in the log:
Unable to start climatisation of vehicle wauzzzfw8r6xxxxxxx: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZFW8RXXXXXXXXX/climater/actions'

My Car:
Audi RS e-tron GT 2024 model Great Britain / EU Electric Vehicle. 

Neither starts the Climate control, though the previous version provided by @lexathon did, but only in Legacy mode, there was no ability to set temperature, so on a cold morning, if the A/C was used before it used the previous setting 🥶

Hope this helps

Kolbi commented:
> 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_account] Initiating Start Climate Control Service...
> 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_connect_account] Sending command to start climate control for vehicle [SNremoved] with settings - Temp(F): None, Temp(C): None, Glass Heating: False, Seat FL: False, Seat FR: False, Seat RL: False, Seat RR: False
> 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_services] Attempting to start climate control with API Level 0 and country DE.
> 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_api] Request initiated: method=POST, url=https://emea.bff.cariad.digital/vehicle/v1/vehicles/[SNremoved]/climatisation/start, data={"action": {"type": "startClimatisation", "settings": {"targetTemperature": 2941, "climatisationWithoutHVpower": true, "heaterSource": "electric", "climaterElementSettings": {"isClimatisationAtUnlock": false, "isMirrorHeatingEnabled": false, "zoneSettings": {"zoneSetting": [{"value": {"isEnabled": false, "position": "frontLeft"}}, {"value": {"isEnabled": false, "position": "frontRight"}}, {"value": {"isEnabled": false, "position": "rearLeft"}}, {"value": {"isEnabled": false, "position": "rearRight"}}]}}}}}, headers={'User-Agent': "Android/4.31.0 (Build 800341641.root project 'myaudi_android'.ext.buildTime) Android/13", 'Host': 'emea.bff.cariad.digital', 'X-App-Version': '4.31.0', 'X-App-Name': 'myAudi', 'Authorization': '', 'Accept-charset': 'UTF-8', 'Content-Type': 'application/json', 'Accept': 'application/json, application/vnd.vwg.mbb.ChargerAction_v1_0_0+xml,application/vnd.volkswagenag.com-error-v1+xml,application/vnd.vwg.mbb.genericError_v1_0_2+xml, application/vnd.vwg.mbb.RemoteStandheizung_v2_0_0+xml, application/vnd.vwg.mbb.genericError_v1_0_2+xml,application/vnd.vwg.mbb.RemoteLockUnlock_v1_0_0+xml,*/*'}, kwargs={}
> 2025-03-19 11:45:35.853 ERROR (MainThread) [custom_components.audiconnect.audi_connect_account] Unable to start climate control of vehicle [SNremoved]. Error: 403, message='Forbidden', url='https://emea.bff.cariad.digital/vehicle/v1/vehicles/[SNremoved]/climatisation/start'
> Traceback (most recent call last):
>   File "/config/custom_components/audiconnect/audi_connect_account.py", line 320, in start_climate_control
>     await self._audi_service.start_climate_control(
>     ...<8 lines>...
>     )
>   File "/config/custom_components/audiconnect/audi_services.py", line 676, in start_climate_control
>     res = await self._api.request(
>           ^^^^^^^^^^^^^^^^^^^^^^^^
>     ...<6 lines>...
>     )
>     ^
>   File "/config/custom_components/audiconnect/audi_api.py", line 79, in request
>     raise ClientResponseError(
>     ...<4 lines>...
>     )
> aiohttp.client_exceptions.ClientResponseError: 403, message='Forbidden', url='https://emea.bff.cariad.digital/vehicle/v1/vehicles/[SNremoved]/climatisation/start'
> 2025-03-19 11:46:00.051 DEBUG (MainThread) [custom_components.audiconnect.audi_connect_account] Sending command to start climatisation to vehicle [SNremoved]
> 2025-03-19 11:46:00.051 DEBUG (MainThread) [custom_components.audiconnect.audi_api] Request initiated: method=POST, url=https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions, data={"action":{"type": "startClimatisation","settings": {"targetTemperature": 2940,"climatisationWithoutHVpower": true,"heaterSource": "electric","climaterElementSettings": {"isClimatisationAtUnlock": false, "isMirrorHeatingEnabled": true,}}}}, headers={'User-Agent': "Android/4.31.0 (Build 800341641.root project 'myaudi_android'.ext.buildTime) Android/13", 'Host': 'emea.bff.cariad.digital', 'X-App-Version': '4.31.0', 'X-App-Name': 'myAudi', 'Authorization': '', 'Accept-charset': 'UTF-8', 'Content-Type': 'application/json', 'Accept': 'application/json, application/vnd.vwg.mbb.ChargerAction_v1_0_0+xml,application/vnd.volkswagenag.com-error-v1+xml,application/vnd.vwg.mbb.genericError_v1_0_2+xml, application/vnd.vwg.mbb.RemoteStandheizung_v2_0_0+xml, application/vnd.vwg.mbb.genericError_v1_0_2+xml,application/vnd.vwg.mbb.RemoteLockUnlock_v1_0_0+xml,*/*'}, kwargs={}
> 2025-03-19 11:46:00.290 ERROR (MainThread) [custom_components.audiconnect.util] Unable to start climatisation of vehicle [SNremoved]: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions'
> 2025-03-19 11:46:07.539 DEBUG (MainThread) [custom_components.audiconnect.audi_connect_account] Sending command to stop climatisation to vehicle [SNremoved]
> 2025-03-19 11:46:07.539 DEBUG (MainThread) [custom_components.audiconnect.audi_api] Request initiated: method=POST, url=https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions, data={"action":{"type": "stopClimatisation"}}, headers={'User-Agent': "Android/4.31.0 (Build 800341641.root project 'myaudi_android'.ext.buildTime) Android/13", 'Host': 'emea.bff.cariad.digital', 'X-App-Version': '4.31.0', 'X-App-Name': 'myAudi', 'Authorization': '', 'Accept-charset': 'UTF-8', 'Content-Type': 'application/json', 'Accept': 'application/json, application/vnd.vwg.mbb.ChargerAction_v1_0_0+xml,application/vnd.volkswagenag.com-error-v1+xml,application/vnd.vwg.mbb.genericError_v1_0_2+xml, application/vnd.vwg.mbb.RemoteStandheizung_v2_0_0+xml, application/vnd.vwg.mbb.genericError_v1_0_2+xml,application/vnd.vwg.mbb.RemoteLockUnlock_v1_0_0+xml,*/*'}, kwargs={}
> 2025-03-19 11:46:07.584 ERROR (MainThread) [custom_components.audiconnect.util] Unable to stop climatisation of vehicle [SNremoved]: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions'

coreywillwhat commented:
> For easier testing I created a manual prelease: https://github.com/audiconnect/audi_connect_ha/releases/tag/v1.11.3 should be able to update via HACS if you choose manually the version to download.
> 
> I sadly can't test it right now.

Love the pre-release idea. I was hesitant after the last discussion with @cdnninja. Anything we need to consider [given that discussion ](https://github.com/audiconnect/audi_connect_ha/discussions/272#discussioncomment-9099171)to be sure we don't accidentally push the beta and final?

coreywillwhat commented:
> So two things. The API level change in configuration doesn't seem to retain its state - not 100% sure if switching to level 1 is being kept somewhere - URL seems right, so I'm assuming it is, but when you reopen configure, it's back at 0.
> 
> The URL matches, but I'm wondering if there's something else going on with the headers that's kicking back a forbidden?

Great catch with the configure menu. You're right, the value is being stored correctly on the backend, but the menu is displaying the default value each time the menu is opened. This is a quick fix, I'll update shortly.

It is likely the headers as you're suggesting. For [your patch ](https://github.com/audiconnect/audi_connect_ha/issues/532#issue-2920527883) that is working for `set_climatisation`, your headers are:
``` python
headers = { "Authorization": "Bearer " + self._bearer_token_json["access_token"] }
```
This is working for you in your patch? Seems odd you are only sending the authorization as the headers. I tried to implement the same logic if your region is "DE" (EU), so unsure why it isn't working if it is in your own code. I tried this for my region (US) and it did not work. I will try using the same method I'm using for US and let you know when its ready for testing.

coreywillwhat commented:
> OK I have just tested with the pre-release, I might have gotten the wrong end of the stick. I am using the "Audi Connect: Start Climate Control" and setting a temperature on the car with my VIN number installed Action to call it, or should I be using Legacy?
> 
> I get in the log still: Unable to start climate control of vehicle wauzzzfw8rXXXXXXXX. Error: 403, message='Forbidden', url='https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZFW8XXXXXXX/climatisation/start'
> 
> ```
> Traceback (most recent call last):
>   File "/config/custom_components/audiconnect/audi_connect_account.py", line 320, in start_climate_control
>     await self._audi_service.start_climate_control(
>     ...<8 lines>...
>     )
>   File "/config/custom_components/audiconnect/audi_services.py", line 676, in start_climate_control
>     res = await self._api.request(
>           ^^^^^^^^^^^^^^^^^^^^^^^^
>     ...<6 lines>...
>     )
>     ^
>   File "/config/custom_components/audiconnect/audi_api.py", line 79, in request
>     raise ClientResponseError(
>     ...<4 lines>...
>     )
> aiohttp.client_exceptions.ClientResponseError: 403, message='Forbidden', url='https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZFW8XXXXXXXX/climatisation/start'
> ```
> 
> On the legacy Action I get this in the log: Unable to start climatisation of vehicle wauzzzfw8r6xxxxxxx: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZFW8RXXXXXXXXX/climater/actions'
> 
> My Car: Audi RS e-tron GT 2024 model Great Britain / EU Electric Vehicle.
> 
> Neither starts the Climate control, though the previous version provided by @lexathon did, but only in Legacy mode, there was no ability to set temperature, so on a cold morning, if the A/C was used before it used the previous setting 🥶
> 
> Hope this helps

You're first statement is correct, this PR is only for `start_climate_control` service, not the legacy action. Thanks for testing. I'm thinking this is related to headers being sent with the request.

coreywillwhat commented:
> > 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_account] Initiating Start Climate Control Service...
> > 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_connect_account] Sending command to start climate control for vehicle [SNremoved] with settings - Temp(F): None, Temp(C): None, Glass Heating: False, Seat FL: False, Seat FR: False, Seat RL: False, Seat RR: False
> > 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_services] Attempting to start climate control with API Level 0 and country DE.
> > 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_api] Request initiated: method=POST, url=[https://emea.bff.cariad.digital/vehicle/v1/vehicles/[SNremoved]/climatisation/start](https://emea.bff.cariad.digital/vehicle/v1/vehicles/%5BSNremoved%5D/climatisation/start), data={"action": {"type": "startClimatisation", "settings": {"targetTemperature": 2941, "climatisationWithoutHVpower": true, "heaterSource": "electric", "climaterElementSettings": {"isClimatisationAtUnlock": false, "isMirrorHeatingEnabled": false, "zoneSettings": {"zoneSetting": [{"value": {"isEnabled": false, "position": "frontLeft"}}, {"value": {"isEnabled": false, "position": "frontRight"}}, {"value": {"isEnabled": false, "position": "rearLeft"}}, {"value": {"isEnabled": false, "position": "rearRight"}}]}}}}}, headers={'User-Agent': "Android/4.31.0 (Build 800341641.root project 'myaudi_android'.ext.buildTime) Android/13", 'Host': 'emea.bff.cariad.digital', 'X-App-Version': '4.31.0', 'X-App-Name': 'myAudi', 'Authorization': '', 'Accept-charset': 'UTF-8', 'Content-Type': 'application/json', 'Accept': 'application/json, application/vnd.vwg.mbb.ChargerAction_v1_0_0+xml,application/vnd.volkswagenag.com-error-v1+xml,application/vnd.vwg.mbb.genericError_v1_0_2+xml, application/vnd.vwg.mbb.RemoteStandheizung_v2_0_0+xml, application/vnd.vwg.mbb.genericError_v1_0_2+xml,application/vnd.vwg.mbb.RemoteLockUnlock_v1_0_0+xml,_/_'}, kwargs={}
> > 2025-03-19 11:45:35.853 ERROR (MainThread) [custom_components.audiconnect.audi_connect_account] Unable to start climate control of vehicle [SNremoved]. Error: 403, message='Forbidden', url='[https://emea.bff.cariad.digital/vehicle/v1/vehicles/[SNremoved]/climatisation/start](https://emea.bff.cariad.digital/vehicle/v1/vehicles/%5BSNremoved%5D/climatisation/start)'
> > Traceback (most recent call last):
> > File "/config/custom_components/audiconnect/audi_connect_account.py", line 320, in start_climate_control
> > await self._audi_service.start_climate_control(
> > ...<8 lines>...
> > )
> > File "/config/custom_components/audiconnect/audi_services.py", line 676, in start_climate_control
> > res = await self._api.request(
> > ^^^^^^^^^^^^^^^^^^^^^^^^
> > ...<6 lines>...
> > )
> > ^
> > File "/config/custom_components/audiconnect/audi_api.py", line 79, in request
> > raise ClientResponseError(
> > ...<4 lines>...
> > )
> > aiohttp.client_exceptions.ClientResponseError: 403, message='Forbidden', url='[https://emea.bff.cariad.digital/vehicle/v1/vehicles/[SNremoved]/climatisation/start](https://emea.bff.cariad.digital/vehicle/v1/vehicles/%5BSNremoved%5D/climatisation/start)'
> > 2025-03-19 11:46:00.051 DEBUG (MainThread) [custom_components.audiconnect.audi_connect_account] Sending command to start climatisation to vehicle [SNremoved]
> > 2025-03-19 11:46:00.051 DEBUG (MainThread) [custom_components.audiconnect.audi_api] Request initiated: method=POST, url=[https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions](https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/%5BSNremoved%5D/climater/actions), data={"action":{"type": "startClimatisation","settings": {"targetTemperature": 2940,"climatisationWithoutHVpower": true,"heaterSource": "electric","climaterElementSettings": {"isClimatisationAtUnlock": false, "isMirrorHeatingEnabled": true,}}}}, headers={'User-Agent': "Android/4.31.0 (Build 800341641.root project 'myaudi_android'.ext.buildTime) Android/13", 'Host': 'emea.bff.cariad.digital', 'X-App-Version': '4.31.0', 'X-App-Name': 'myAudi', 'Authorization': '', 'Accept-charset': 'UTF-8', 'Content-Type': 'application/json', 'Accept': 'application/json, application/vnd.vwg.mbb.ChargerAction_v1_0_0+xml,application/vnd.volkswagenag.com-error-v1+xml,application/vnd.vwg.mbb.genericError_v1_0_2+xml, application/vnd.vwg.mbb.RemoteStandheizung_v2_0_0+xml, application/vnd.vwg.mbb.genericError_v1_0_2+xml,application/vnd.vwg.mbb.RemoteLockUnlock_v1_0_0+xml,_/_'}, kwargs={}
> > 2025-03-19 11:46:00.290 ERROR (MainThread) [custom_components.audiconnect.util] Unable to start climatisation of vehicle [SNremoved]: 404, message='Not Found', url='[https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions](https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/%5BSNremoved%5D/climater/actions)'
> > 2025-03-19 11:46:07.539 DEBUG (MainThread) [custom_components.audiconnect.audi_connect_account] Sending command to stop climatisation to vehicle [SNremoved]
> > 2025-03-19 11:46:07.539 DEBUG (MainThread) [custom_components.audiconnect.audi_api] Request initiated: method=POST, url=[https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions](https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/%5BSNremoved%5D/climater/actions), data={"action":{"type": "stopClimatisation"}}, headers={'User-Agent': "Android/4.31.0 (Build 800341641.root project 'myaudi_android'.ext.buildTime) Android/13", 'Host': 'emea.bff.cariad.digital', 'X-App-Version': '4.31.0', 'X-App-Name': 'myAudi', 'Authorization': '', 'Accept-charset': 'UTF-8', 'Content-Type': 'application/json', 'Accept': 'application/json, application/vnd.vwg.mbb.ChargerAction_v1_0_0+xml,application/vnd.volkswagenag.com-error-v1+xml,application/vnd.vwg.mbb.genericError_v1_0_2+xml, application/vnd.vwg.mbb.RemoteStandheizung_v2_0_0+xml, application/vnd.vwg.mbb.genericError_v1_0_2+xml,application/vnd.vwg.mbb.RemoteLockUnlock_v1_0_0+xml,_/_'}, kwargs={}
> > 2025-03-19 11:46:07.584 ERROR (MainThread) [custom_components.audiconnect.util] Unable to stop climatisation of vehicle [SNremoved]: 404, message='Not Found', url='[https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions](https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/%5BSNremoved%5D/climater/actions)'

Interesting your authorization is NULL in the request...

Kolbi commented:
> > > 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_account] Initiating Start Climate Control Service...
> > > 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_connect_account] Sending command to start climate control for vehicle [SNremoved] with settings - Temp(F): None, Temp(C): None, Glass Heating: False, Seat FL: False, Seat FR: False, Seat RL: False, Seat RR: False
> > > 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_services] Attempting to start climate control with API Level 0 and country DE.
> > > 2025-03-19 11:45:35.735 DEBUG (MainThread) [custom_components.audiconnect.audi_api] Request initiated: method=POST, url=[https://emea.bff.cariad.digital/vehicle/v1/vehicles/[SNremoved]/climatisation/start](https://emea.bff.cariad.digital/vehicle/v1/vehicles/%5BSNremoved%5D/climatisation/start), data={"action": {"type": "startClimatisation", "settings": {"targetTemperature": 2941, "climatisationWithoutHVpower": true, "heaterSource": "electric", "climaterElementSettings": {"isClimatisationAtUnlock": false, "isMirrorHeatingEnabled": false, "zoneSettings": {"zoneSetting": [{"value": {"isEnabled": false, "position": "frontLeft"}}, {"value": {"isEnabled": false, "position": "frontRight"}}, {"value": {"isEnabled": false, "position": "rearLeft"}}, {"value": {"isEnabled": false, "position": "rearRight"}}]}}}}}, headers={'User-Agent': "Android/4.31.0 (Build 800341641.root project 'myaudi_android'.ext.buildTime) Android/13", 'Host': 'emea.bff.cariad.digital', 'X-App-Version': '4.31.0', 'X-App-Name': 'myAudi', 'Authorization': '', 'Accept-charset': 'UTF-8', 'Content-Type': 'application/json', 'Accept': 'application/json, application/vnd.vwg.mbb.ChargerAction_v1_0_0+xml,application/vnd.volkswagenag.com-error-v1+xml,application/vnd.vwg.mbb.genericError_v1_0_2+xml, application/vnd.vwg.mbb.RemoteStandheizung_v2_0_0+xml, application/vnd.vwg.mbb.genericError_v1_0_2+xml,application/vnd.vwg.mbb.RemoteLockUnlock_v1_0_0+xml,_/_'}, kwargs={}
> > > 2025-03-19 11:45:35.853 ERROR (MainThread) [custom_components.audiconnect.audi_connect_account] Unable to start climate control of vehicle [SNremoved]. Error: 403, message='Forbidden', url='[https://emea.bff.cariad.digital/vehicle/v1/vehicles/[SNremoved]/climatisation/start](https://emea.bff.cariad.digital/vehicle/v1/vehicles/%5BSNremoved%5D/climatisation/start)'
> > > Traceback (most recent call last):
> > > File "/config/custom_components/audiconnect/audi_connect_account.py", line 320, in start_climate_control
> > > await self._audi_service.start_climate_control(
> > > ...<8 lines>...
> > > )
> > > File "/config/custom_components/audiconnect/audi_services.py", line 676, in start_climate_control
> > > res = await self._api.request(
> > > ^^^^^^^^^^^^^^^^^^^^^^^^
> > > ...<6 lines>...
> > > )
> > > ^
> > > File "/config/custom_components/audiconnect/audi_api.py", line 79, in request
> > > raise ClientResponseError(
> > > ...<4 lines>...
> > > )
> > > aiohttp.client_exceptions.ClientResponseError: 403, message='Forbidden', url='[https://emea.bff.cariad.digital/vehicle/v1/vehicles/[SNremoved]/climatisation/start](https://emea.bff.cariad.digital/vehicle/v1/vehicles/%5BSNremoved%5D/climatisation/start)'
> > > 2025-03-19 11:46:00.051 DEBUG (MainThread) [custom_components.audiconnect.audi_connect_account] Sending command to start climatisation to vehicle [SNremoved]
> > > 2025-03-19 11:46:00.051 DEBUG (MainThread) [custom_components.audiconnect.audi_api] Request initiated: method=POST, url=[https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions](https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/%5BSNremoved%5D/climater/actions), data={"action":{"type": "startClimatisation","settings": {"targetTemperature": 2940,"climatisationWithoutHVpower": true,"heaterSource": "electric","climaterElementSettings": {"isClimatisationAtUnlock": false, "isMirrorHeatingEnabled": true,}}}}, headers={'User-Agent': "Android/4.31.0 (Build 800341641.root project 'myaudi_android'.ext.buildTime) Android/13", 'Host': 'emea.bff.cariad.digital', 'X-App-Version': '4.31.0', 'X-App-Name': 'myAudi', 'Authorization': '', 'Accept-charset': 'UTF-8', 'Content-Type': 'application/json', 'Accept': 'application/json, application/vnd.vwg.mbb.ChargerAction_v1_0_0+xml,application/vnd.volkswagenag.com-error-v1+xml,application/vnd.vwg.mbb.genericError_v1_0_2+xml, application/vnd.vwg.mbb.RemoteStandheizung_v2_0_0+xml, application/vnd.vwg.mbb.genericError_v1_0_2+xml,application/vnd.vwg.mbb.RemoteLockUnlock_v1_0_0+xml,_/_'}, kwargs={}
> > > 2025-03-19 11:46:00.290 ERROR (MainThread) [custom_components.audiconnect.util] Unable to start climatisation of vehicle [SNremoved]: 404, message='Not Found', url='[https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions](https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/%5BSNremoved%5D/climater/actions)'
> > > 2025-03-19 11:46:07.539 DEBUG (MainThread) [custom_components.audiconnect.audi_connect_account] Sending command to stop climatisation to vehicle [SNremoved]
> > > 2025-03-19 11:46:07.539 DEBUG (MainThread) [custom_components.audiconnect.audi_api] Request initiated: method=POST, url=[https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions](https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/%5BSNremoved%5D/climater/actions), data={"action":{"type": "stopClimatisation"}}, headers={'User-Agent': "Android/4.31.0 (Build 800341641.root project 'myaudi_android'.ext.buildTime) Android/13", 'Host': 'emea.bff.cariad.digital', 'X-App-Version': '4.31.0', 'X-App-Name': 'myAudi', 'Authorization': '', 'Accept-charset': 'UTF-8', 'Content-Type': 'application/json', 'Accept': 'application/json, application/vnd.vwg.mbb.ChargerAction_v1_0_0+xml,application/vnd.volkswagenag.com-error-v1+xml,application/vnd.vwg.mbb.genericError_v1_0_2+xml, application/vnd.vwg.mbb.RemoteStandheizung_v2_0_0+xml, application/vnd.vwg.mbb.genericError_v1_0_2+xml,application/vnd.vwg.mbb.RemoteLockUnlock_v1_0_0+xml,_/_'}, kwargs={}
> > > 2025-03-19 11:46:07.584 ERROR (MainThread) [custom_components.audiconnect.util] Unable to stop climatisation of vehicle [SNremoved]: 404, message='Not Found', url='[https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/[SNremoved]/climater/actions](https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/%5BSNremoved%5D/climater/actions)'
> 
> Interesting your authorization is NULL in the request...

I deleted the bearer Token out of the log :)

lexathon commented:
and I can confirm that the data seems to pass through effectively.

coreywillwhat commented:
> and I can confirm that the data seems to pass through effectively.

Updated this PR to fix the headers for EU using your working code. I'm still not convinced this is Region based and not API based, but we wont be able to tell unless someone with an older vehicle can confirm from EU. 

Also fixed the options menu to load the saved value first if available, else default.

coreywillwhat commented:
> @Kolbi great if you can do another manual prerelease - makes it easier to pull. thx.

Just posted v1.11.4 - Pre-release #539

lexathon commented:
superstar. will pull it now and check. thanks for all your work on this.

coreywillwhat commented:
> Weirdly, I got this from the logs
> 
> 
> 
> Note API 0 logged but I definitely set it to 1 and also just checked again and it's set to 1
> 
> 
> 
> ```
> 
> 2025-03-19 17:41:30.589 DEBUG (MainThread) [custom_components.audiconnect.audi_services] Attempting to start climate control with API Level 0 and country DE.
> 
> 2025-03-19 17:41:30.589 DEBUG (MainThread) [custom_components.audiconnect.audi_api] Request initiated: method=POST, url=https://emea.bff.cariad.digital/vehicle/v1/vehicles/**REDACTED**/climatisation/start, data={"action": {"type": "startClimatisation", "settings": {"targetTemperature": 2921, "climatisationWithoutHVpower": true, "heaterSource": "electric", "climaterElementSettings": {"isClimatisationAtUnlock": false, "isMirrorHeatingEnabled": false, "zoneSettings": {"zoneSetting": [{"value": {"isEnabled": false, "position": "frontLeft"}}, {"value": {"isEnabled": false, "position": "frontRight"}}, {"value": {"isEnabled": false, "position": "rearLeft"}}, {"value": {"isEnabled": false, "position": "rearRight"}}]}}}}}, headers={'Authorization': 'Bearer **REDACTED**'}, kwargs={}
> 
> 2025-03-19 17:41:30.780 DEBUG (MainThread) [custom_components.audiconnect.audi_connect_account] Successfully started climate control of vehicle wauzzzfz6sp040775
> 
> ```

Interesting it worked, the data structure used in the request is API 0 FYI.

shuntfield commented:
This error originated from a custom integration.

Logger: custom_components.audiconnect.util
Source: custom_components/audiconnect/util.py:26
integration: Audi Connect (documentation, issues)
First occurred: 18:30:29 (1 occurrences)
Last logged: 18:30:29

Unable to start climatisation of vehicle wauzzzfw8XXXXXXX: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZFWXXXXXXX/climater/actions'

I've just installed the pre release, and still I am getting this error perhaps I should try and remove and reload it?

bla3k commented:
I just updated to the pre-release version and still get the same error message.
It doesn't matter which API I select.

```
Unable to start climatisation of vehicle WAUZZZxxxxxxxx: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZxxxxxxxx/climater/actions'

Unable to start pre-heater of vehicle WAUZZZxxxxxxxx: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/rs/v1/Audi/DE/vehicles/WAUZZZxxxxxxxx/action'
```

Another thing I noticed is that with 1.11.4, logging in is no longer possible if I completely remove the add-on and reinstall with the pre-release. Installing and configuring with 1.11.2, then updating to the pre-release works.

Model: Audi A4 2020
Region: DE/EU

Thank you for your work!

coreywillwhat commented:
> This error originated from a custom integration.
> 
> 
> 
> Logger: custom_components.audiconnect.util
> 
> Source: custom_components/audiconnect/util.py:26
> 
> integration: Audi Connect (documentation, issues)
> 
> First occurred: 18:30:29 (1 occurrences)
> 
> Last logged: 18:30:29
> 
> 
> 
> Unable to start climatisation of vehicle wauzzzfw8XXXXXXX: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZFWXXXXXXX/climater/actions'
> 
> 
> 
> I've just installed the pre release, and still I am getting this error perhaps I should try and remove and reload it?
> 
> 
> 
> 

Is this the `start climate control` service or the legacy `execute vehicle action` service?

shuntfield commented:
@coreywillwhat Legacy service.

I completely removed the integration and readded the last pre release now, but I can't log in "Invalid Credentials"

It might be I have locked myself out for a hour, so I will try again shortly 😁 

Thanks for all your efforts though much appreciated 👍🏻

coreywillwhat commented:
I won't be working on any legacy services. Hopefully replace them all with individual, specialized services. This is only for `start_climate_control` specifically.

Another user said downgrading to 1.11.2 allowed them to login, then upgrade to 1.11.4.

shuntfield commented:
> I won't be working on any legacy services. Hopefully replace them all with individual, specialized services. This is only for `start_climate_control` specifically.
> 
> Another user said downgrading to 1.11.2 allowed them to login, then upgrade to 1.11.4.

Yup that worked downgrade login then upgrade 👍🏻, and its working without generating an error to the car too, on the new non-legacy action 🥳. I think I might have sent too many requests at the moment and I am getting a Timeout Error in the Audi App itself for starting climate control, but I think this might be a local problem. 

![image](https://github.com/user-attachments/assets/5fdeeb47-ec3d-47c0-a330-093e38babe3c)

I also had to ensure I selected the Glass Surface and seat heaters, (switched to off/false) even though I didn't need them on for it to work.

Great stuff so far thanks!

coreywillwhat commented:
@bla3k @shuntfield I think I know the source of the login issue, I will make changes and will let you know.

@shuntfield can you expand on the following? 
> I also had to ensure I selected the Glass Surface and seat heaters, (switched to off/false) even though I didn't need them on for it to work.

Default values should be passed if nothing is entered, so these selections should have no impact. Could this have to do with too many commands being sent as you suggested?

shuntfield commented:
> @bla3k @shuntfield I think I know the source of the login issue, I will make changes and will let you know.
> 
> @shuntfield can you expand on the following?
> 
> > I also had to ensure I selected the Glass Surface and seat heaters, (switched to off/false) even though I didn't need them on for it to work.
> 
> Default values should be passed if nothing is entered, so these selections should have no impact. Could this have to do with too many commands being sent as you suggested?

![image](https://github.com/user-attachments/assets/99c611be-a9c5-4506-9491-f52cecc7357b)

Yup, have a look at the pic, I think I need to have them ticked or defined, and switched off. If they are not in there ie unticked then it looks like it doesn't work, at least the command is sent but nothing is happened.

Hope that makes sense!

coreywillwhat commented:
Fixed the `invalid credentials` issue during initial configuration flow. It was far from benign. Thanks for finding @bla3k & @shuntfield!

Please retest if you can and ensure no other issues during clean install, `CONFIGURE` options update, and `start_climate_control` service. I think we're getting close with this one. Thanks for all your help!

See release for more details: [v1.11.5](https://github.com/audiconnect/audi_connect_ha/releases/tag/v1.11.5)

Kolbi commented on custom_components/audiconnect/audi_services.py:
```diff
@@ -591,69 +605,153 @@ async def start_climate_control(
+            if temp_f is not None:
+                target_temperature = int(((temp_f - 32) * (5 / 9)) * 10 + 2731)
+            elif temp_c is not None:
+                target_temperature = int(temp_c * 10 + 2731)
```
I need to check if really only integers are allowed. As my normal temperature is set to 21.5°C :) so I would assume this should also be possible via service. Need to test.

We maybe need .0 and .5 steps.

coreywillwhat replied:
Oh yeah I hadn't thought about this, and don't think anyone has raised this yet. Maybe an enhancement after we get this merged.

bla3k commented:
> Please retest if you can and ensure no other issues during clean install, CONFIGURE options update, and start_climate_control service. I think we're getting close with this one. Thanks for all your help!

Yes, I can confirm that logging in with a clean installation (1.11.5) works.

What I don't know is whether my car even supports the new clima control service. In the app, I can only activate preheating/ventilation. The legacy service no longer works for me. Is there any information about this?

This is the current error message when I use `start_climate_control`.

```
2025-03-20 07:37:07.672 ERROR (MainThread) [custom_components.audiconnect.audi_connect_account] Unable to start climate control of vehicle wauzzzfxxxxxxxxxx. Error: 404, message='Not Found', url='https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZFxxxxxxxxxx/climatisation/start'
Traceback (most recent call last):
  File "/config/custom_components/audiconnect/audi_connect_account.py", line 320, in start_climate_control
    await self._audi_service.start_climate_control(
    ...<8 lines>...
    )
  File "/config/custom_components/audiconnect/audi_services.py", line 681, in start_climate_control
    res = await self._api.request(
          ^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
    )
    ^
  File "/config/custom_components/audiconnect/audi_api.py", line 79, in request
    raise ClientResponseError(
    ...<4 lines>...
    )
aiohttp.client_exceptions.ClientResponseError: 404, message='Not Found', url='https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZFxxxxxxxxxx/climatisation/start'
```

shuntfield commented:
I'm getting a timeout error at the Audi app each time a command is sent, I thought it might have been a combination of bad data signal to the car and the amount of times I had tried it yesterday, but this morning, I am getting the same thing.

![image](https://github.com/user-attachments/assets/9543d42c-e8ca-4e92-b52b-2f47581637b0)

I will try with the latest version, its like as if it is getting the command but something is additionally stopping it.

coreywillwhat commented:
> > Please retest if you can and ensure no other issues during clean install, CONFIGURE options update, and start_climate_control service. I think we're getting close with this one. Thanks for all your help!
> 
> Yes, I can confirm that logging in with a clean installation (1.11.5) works.
> 
> What I don't know is whether my car even supports the new clima control service. In the app, I can only activate preheating/ventilation. The legacy service no longer works for me. Is there any information about this?
> 
> This is the current error message when I use `start_climate_control`.
> 
> ```
> 2025-03-20 07:37:07.672 ERROR (MainThread) [custom_components.audiconnect.audi_connect_account] Unable to start climate control of vehicle wauzzzfxxxxxxxxxx. Error: 404, message='Not Found', url='https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZFxxxxxxxxxx/climatisation/start'
> Traceback (most recent call last):
>   File "/config/custom_components/audiconnect/audi_connect_account.py", line 320, in start_climate_control
>     await self._audi_service.start_climate_control(
>     ...<8 lines>...
>     )
>   File "/config/custom_components/audiconnect/audi_services.py", line 681, in start_climate_control
>     res = await self._api.request(
>           ^^^^^^^^^^^^^^^^^^^^^^^^
>     ...<6 lines>...
>     )
>     ^
>   File "/config/custom_components/audiconnect/audi_api.py", line 79, in request
>     raise ClientResponseError(
>     ...<4 lines>...
>     )
> aiohttp.client_exceptions.ClientResponseError: 404, message='Not Found', url='https://emea.bff.cariad.digital/vehicle/v1/vehicles/WAUZZZFxxxxxxxxxx/climatisation/start'
> ```

Thanks for retesting the login issue!
Unfortunately we can only mimic actions available in the myAudi app for your vehicle. We could work on an enhancement for adding ventilation, but that is outside the scope of this PR.

coreywillwhat commented:
@Kolbi I will work on finalizing this PR with documentation. I don't think there are any outstanding issues which should prevent this from being merged.

blacknell commented:
I just installed 1.11.5 on my other HA instance and I'm getting invalid credentials (using DE) - repeatedly

blacknell commented:
> I just installed 1.11.5 on my other HA instance and I'm getting invalid credentials (using DE) - repeatedly

This also failed in the 1.11.2 release but eventually succeeded. I then re-downloaded the 1.15.5 version.

warrenkerrigan commented:
I have installed 1.11.5 via HACS, and rebooted my HA Yellow twice. I am in the DE (EU) region, and I have tried it on two cars a MY24 Q8 E-Tron and and MY23/24 E-Tron. Both fail.

Interestingly, after the second reboot, the Q4 merged into the Q8 device! :) 

I also do not see the API level option in the Configure modal.

NOTE: /WAUZZZXXXXXXXXXXX = Q8 and WAUZZZYYYYYYYYYYY = Q4

**Q8 E-Tron MY24**
NOTE: it appears that both cars now appear under the same entity, as well as one of the cars remaning a solo flyer.

`Logger: custom_components.audiconnect_backup.audi_connect_account
Source: custom_components/audiconnect_backup/audi_connect_account.py:314
integration: audiconnect_backup
First occurred: 16:21:34 (2 occurrences)
Last logged: 16:23:40

Unable to start climate control of vehicle wauzzzYYYYYYYYYYY. Error: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZYYYYYYYYYYY/climater/actions'
Unable to start climate control of vehicle wauzzzXXXXXXXXXXX. Error: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZXXXXXXXXXXX/climater/actions'
Traceback (most recent call last):
  File "/config/custom_components/audiconnect_backup/audi_connect_account.py", line 314, in start_climate_control
    await self._audi_service.start_climate_control(
    ...<8 lines>...
    )
  File "/config/custom_components/audiconnect_backup/audi_services.py", line 630, in start_climate_control
    res = await self._api.request(
          ^^^^^^^^^^^^^^^^^^^^^^^^
    ...<9 lines>...
    )
    ^
  File "/config/custom_components/audiconnect_backup/audi_api.py", line 67, in request
    raise ClientResponseError(
    ...<4 lines>...
    )
aiohttp.client_exceptions.ClientResponseError: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZXXXXXXXXXXX/climater/actions'`

**Q4 E-Tron MY23**
`Logger: custom_components.audiconnect_backup.audi_connect_account
Source: custom_components/audiconnect_backup/audi_connect_account.py:314
integration: audiconnect_backup
First occurred: 16:21:34 (1 occurrences)
Last logged: 16:21:34

Unable to start climate control of vehicle wauzzzYYYYYYYYYYY. Error: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZYYYYYYYYYYY/climater/actions'
Traceback (most recent call last):
  File "/config/custom_components/audiconnect_backup/audi_connect_account.py", line 314, in start_climate_control
    await self._audi_service.start_climate_control(
    ...<8 lines>...
    )
  File "/config/custom_components/audiconnect_backup/audi_services.py", line 630, in start_climate_control
    res = await self._api.request(
          ^^^^^^^^^^^^^^^^^^^^^^^^
    ...<9 lines>...
    )
    ^
  File "/config/custom_components/audiconnect_backup/audi_api.py", line 67, in request
    raise ClientResponseError(
    ...<4 lines>...
    )
aiohttp.client_exceptions.ClientResponseError: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZYYYYYYYYYYY/climater/actions'`

I have to run out now, but I have attached some screenshots.
![Screenshot 2025-03-20 at 16 16 20](https://github.com/user-attachments/assets/d252f39a-43f9-4302-80a4-e594bf1fe162)
![Screenshot 2025-03-20 at 16 16 52](https://github.com/user-attachments/assets/11e339ac-8f54-45e1-acd9-99bffa04af4e)
![Screenshot 2025-03-20 at 16 22 57](https://github.com/user-attachments/assets/38116cfe-c892-4f40-a710-175f2801c360)
![Screenshot 2025-03-20 at 16 39 19](https://github.com/user-attachments/assets/e63fe321-264d-4f39-98c1-8305cdcf88b9)

coreywillwhat commented:
Thanks for testing this out @warrenkerrigan !

By the looks of your screenshots and the logs, it doesn't look like the update has taken. There seems to be a local issue with your instance by the looks of the following in the logs:
- This suggests the update failed and the integration is running from a backup
``` python
integration: audiconnect_backup
```
- The URL in the log below does not exist in the code anymore and therefore can't be from version 1.11.5
``` python
Unable to start climate control of vehicle wauzzzXXXXXXXXXXX. Error: 404, message='Not Found', url='https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/climatisation/v1/Audi/DE/vehicles/WAUZZZXXXXXXXXXXX/climater/actions'
```
- Your screenshot shows v1.10.4 in the integration menu. This beta release should still display 1.11.2 as the base in the integration menu. See `manifest.json` in v1.11.5:
``` python
{
  "domain": "audiconnect",
  "name": "Audi Connect",
  "codeowners": ["@arjenvrh"],
  "config_flow": true,
  "documentation": "https://github.com/audiconnect/audi_connect_ha",
  "integration_type": "hub",
  "iot_class": "cloud_polling",
  "issue_tracker": "https://github.com/audiconnect/audi_connect_ha/issues",
  "loggers": ["audiconnect"],
  "requirements": ["beautifulsoup4"],
  "version": "1.11.2"
}
```
From your screenshot:
![image](https://github.com/user-attachments/assets/9fd63eae-ff5c-47f3-a51d-8adad0a74669)


I'd suggest you remove the integration entirely, restart HA, and perform a clean Audi Connect configuration.

coreywillwhat commented:
@blacknell Was this a clean install or are you coming from a previous version? I haven't run into this issue after it was fixed in v1.11.4. 

Note after updating, you may have some old cached data. Suggest using Ctrl+F5 on your browser after updates to be sure you get a fresh load of all the data.

```
In most web browsers, pressing Ctrl + F5 (or Command + Shift + R on Mac) performs a "hard refresh" or "force reload," which bypasses the browser's cache and reloads the page from the server, ensuring you see the most up-to-date version. 
Here's a more detailed explanation:

    What it does:
    Ctrl + F5 forces the browser to ignore its cached version of a webpage and fetch all the page's resources (like HTML, CSS, and JavaScript) directly from the web server. 

Why use it?
This is useful when you suspect a cached version of a page is preventing you from seeing the latest updates or when troubleshooting issues related to cached content. 
Difference from F5:
Simply pressing F5 (or Ctrl + R) performs a regular refresh, which often reloads the page from the browser's cache, potentially showing an outdated version. 
Cross-browser compatibility:
This shortcut works similarly across most web browsers, including Chrome, Firefox, Edge, and Safari. 
```

warrenkerrigan commented:
OK. Thanks for the tip. You were spot on! 
I left an old backup I was messing about with, in that folder.

Removed integration and re-added. 

I have tested both the Q4 and the Q8 in API 1, and the start climate works. API level 0 does not, in either case.

coreywillwhat commented:
Excellent! Thanks for testing and reporting. Glad it's working now.

coreywillwhat commented:
@Kolbi ready for review

Kolbi commented:
Sorry everyone but maybe you need to manually update in HACS to latest release 1.11.3 as we did some mistake with the pre-release versioning...

blacknell commented:
> @blacknell Was this a clean install or are you coming from a previous version? I haven't run into this issue after it was fixed in v1.11.4.
> 
> Note after updating, you may have some old cached data. Suggest using Ctrl+F5 on your browser after updates to be sure you get a fresh load of all the data.
> 
> ```
> In most web browsers, pressing Ctrl + F5 (or Command + Shift + R on Mac) performs a "hard refresh" or "force reload," which bypasses the browser's cache and reloads the page from the server, ensuring you see the most up-to-date version. 
> Here's a more detailed explanation:
> 
>     What it does:
>     Ctrl + F5 forces the browser to ignore its cached version of a webpage and fetch all the page's resources (like HTML, CSS, and JavaScript) directly from the web server. 
> 
> Why use it?
> This is useful when you suspect a cached version of a page is preventing you from seeing the latest updates or when troubleshooting issues related to cached content. 
> Difference from F5:
> Simply pressing F5 (or Ctrl + R) performs a regular refresh, which often reloads the page from the browser's cache, potentially showing an outdated version. 
> Cross-browser compatibility:
> This shortcut works similarly across most web browsers, including Chrome, Firefox, Edge, and Safari. 
> ```

This was a clean install on my 2nd instance. It failed several times but eventually worked (I had gone back to 1.11.2 though). I think this is a red herring related to some oddity as you suggest.

coreywillwhat commented:
Got it. Thanks for reporting back!

blacknell commented:
@coreywillwhat don't you think we should have an action to turn off climate control? If I add a button on my dashboards to turn on, I'd like to be able to toggle it off (for when I inevitably accidentally turn it on!)

pergolafabio commented:
Climate off will come in future too

coreywillwhat commented:
> @coreywillwhat don't you think we should have an action to turn off climate control? If I add a button on my dashboards to turn on, I'd like to be able to toggle it off (for when I inevitably accidentally turn it on!)

Yeah you're absolutely right. Just a matter of time. Planning to fix it this week, hopefully before the next auto-release.

blacknell commented:
@coreywillwhat also, i noticed that the state of climatisation doesn't update in HA in a timely manner after turning on climate. When i execute a refresh cloud data action, then it updates. I thought I saw in the logs that it does this refresh as part of the set climate call but perhaps the delay needs to be a bit longer.

a workaround would be to put a wait and refesh step into my automation but wondered if you had any opinions

coreywillwhat commented:
@blacknell the workaround you suggests already exists until i get the check request succeeded fixed. See my comment [here](https://github.com/audiconnect/audi_connect_ha/issues/450#issuecomment-2742101648).
